stablehlo/tests/interpret_iota.mlir

```diff
@@ -0,0 +1,1213 @@
+}
+
+// CHECK-LABEL: Evaluated results of function: iota_op_test_si16_dim_1
+func.func @iota_op_test_si16_dim_1() -> tensor<3x4xi16> {
+  %0 = stablehlo.iota dim = 1 : tensor<3x4xi16>
+  func.return %0 : tensor<3x4xi16>
+  // CHECK-NEXT: tensor<3x4xi16>
+  // CHECK-NEXT: 0 : i16
+  // CHECK-NEXT: 1 : i16
+  // CHECK-NEXT: 2 : i16
+  // CHECK-NEXT: 3 : i16
+  // CHECK-NEXT: 0 : i16
+  // CHECK-NEXT: 1 : i16
+  // CHECK-NEXT: 2 : i16
+  // CHECK-NEXT: 3 : i16
+  // CHECK-NEXT: 0 : i16
+  // CHECK-NEXT: 1 : i16
+  // CHECK-NEXT: 2 : i16
+  // CHECK-NEXT: 3 : i16
+}
+
+// CHECK-LABEL: Evaluated results of function: iota_op_test_si32_dim_0
+func.func @iota_op_test_si32_dim_0() -> tensor<3x4xi32> {
+  %0 = stablehlo.iota dim = 0 : tensor<3x4xi32>
+  func.return %0 : tensor<3x4xi32>
+  // CHECK-NEXT: tensor<3x4xi32>
+  // CHECK-NEXT: 0 : i32
+  // CHECK-NEXT: 0 : i32
+  // CHECK-NEXT: 0 : i32
+  // CHECK-NEXT: 0 : i32
+  // CHECK-NEXT: 1 : i32
+  // CHECK-NEXT: 1 : i32
+  // CHECK-NEXT: 1 : i32
+  // CHECK-NEXT: 1 : i32
+  // CHECK-NEXT: 2 : i32
+  // CHECK-NEXT: 2 : i32
+  // CHECK-NEXT: 2 : i32
+  // CHECK-NEXT: 2 : i32
+}
+
+// CHECK-LABEL: Evaluated results of function: iota_op_test_si32_dim_1
+func.func @iota_op_test_si32_dim_1() -> tensor<3x4xi32> {
+  %0 = stablehlo.iota dim = 1 : tensor<3x4xi32>
+  func.return %0 : tensor<3x4xi32>
+  // CHECK-NEXT: tensor<3x4xi32>
+  // CHECK-NEXT: 0 : i32
+  // CHECK-NEXT: 1 : i32
+  // CHECK-NEXT: 2 : i32
+  // CHECK-NEXT: 3 : i32
+  // CHECK-NEXT: 0 : i32
+  // CHECK-NEXT: 1 : i32
+  // CHECK-NEXT: 2 : i32
+  // CHECK-NEXT: 3 : i32
+  // CHECK-NEXT: 0 : i32
+  // CHECK-NEXT: 1 : i32
+  // CHECK-NEXT: 2 : i32
+  // CHECK-NEXT: 3 : i32
+}
+
+// CHECK-LABEL: Evaluated results of function: iota_op_test_si64_dim_0
+func.func @iota_op_test_si64_dim_0() -> tensor<3x4xi64> {
+  %0 = stablehlo.iota dim = 0 : tensor<3x4xi64>
+  func.return %0 : tensor<3x4xi64>
+  // CHECK-NEXT: tensor<3x4xi64>
+  // CHECK-NEXT: 0 : i64
+  // CHECK-NEXT: 0 : i64
+  // CHECK-NEXT: 0 : i64
+  // CHECK-NEXT: 0 : i64
+  // CHECK-NEXT: 1 : i64
+  // CHECK-NEXT: 1 : i64
+  // CHECK-NEXT: 1 : i64
+  // CHECK-NEXT: 1 : i64
+  // CHECK-NEXT: 2 : i64
+  // CHECK-NEXT: 2 : i64
+  // CHECK-NEXT: 2 : i64
+  // CHECK-NEXT: 2 : i64
+}
+
+// CHECK-LABEL: Evaluated results of function: iota_op_test_si64_dim_1
+func.func @iota_op_test_si64_dim_1() -> tensor<3x4xi64> {
+  %0 = stablehlo.iota dim = 1 : tensor<3x4xi64>
+  func.return %0 : tensor<3x4xi64>
+  // CHECK-NEXT: tensor<3x4xi64>
+  // CHECK-NEXT: 0 : i64
+  // CHECK-NEXT: 1 : i64
+  // CHECK-NEXT: 2 : i64
+  // CHECK-NEXT: 3 : i64
+  // CHECK-NEXT: 0 : i64
+  // CHECK-NEXT: 1 : i64
+  // CHECK-NEXT: 2 : i64
+  // CHECK-NEXT: 3 : i64
+  // CHECK-NEXT: 0 : i64
+  // CHECK-NEXT: 1 : i64
+  // CHECK-NEXT: 2 : i64
+  // CHECK-NEXT: 3 : i64
+}
+
+// CHECK-LABEL: Evaluated results of function: iota_op_test_ui64_dim_0
+func.func @iota_op_test_ui64_dim_0() -> tensor<3x4x5x3xui64> {
```

Comment:
How about keeping the dimension sizes <= 3 and rank == 3; an example shape could be 2x3x2, which is manageable w.r.t the element prints and still can demonstrates a relative complex case. The output here is too lengthy to follow.

Reply:
same for other `3x4x5x3` cases below.